docs/gen-ai/gen-ai-events.md

```diff
@@ -0,0 +1,379 @@
+
+This example covers the following scenario:
+
+1. Application requests chat completion from OpenAI GPT-4 model and provides a function definition.
+
+   - Application provides the following prompt:
+     - User message: `How to instrument GenAI library with OTel?`
+   - Application defines a tool (a function) names `get_link_to_otel_semconv` with single string argument named `semconv`
+
+2. The model responds with a tool call request which application executes
+3. The application requests chat completion again now with the tool execution result
+
+Here's the telemetry generated for each step in this scenario:
+
+1. Chat completion resulting in a tool call.
+
+   |   Attribute name    |                     Value                             |
+   |---------------------|-------------------------------------------------------|
+   | Span name           | `"chat gpt-4"`                             |
+   | `gen_ai.system`     | `"openai"`                                            |
+   | `gen_ai.request.model`| `"gpt-4"`                                           |
+   | `gen_ai.request.max_tokens`| `200`                                          |
+   | `gen_ai.request.top_p`| `1.0`                                               |
+   | `gen_ai.response.id`| `"chatcmpl-9J3uIL87gldCFtiIbyaOvTeYBRA3l"`            |
+   | `gen_ai.response.model`| `"gpt-4-0613"`                                     |
+   | `gen_ai.usage.output_tokens`| `17`                                      |
+   | `gen_ai.usage.input_tokens`| `47`                                          |
+   | `gen_ai.response.finish_reasons`| `["tool_calls"]`                             |
+
+   Events parented to this span:
+
+   - `gen_ai.user.message` (not reported when capturing content is disabled)
+
+     |   Property          |                     Value                             |
+     |---------------------|-------------------------------------------------------|
+     | `gen_ai.system`     | `"openai"`                                            |
+     | Event body       | `{"content":"How to instrument GenAI library with OTel?"}` |
+
+   - `gen_ai.choice`
+
+     |   Property          |                     Value                             |
+     |---------------------|-------------------------------------------------------|
+     | `gen_ai.system`     | `"openai"`                                            |
+     | Event body (with content)    | `{"index":0,"finish_reason":"tool_calls","message":{"tool_calls":[{"id":"call_VSPygqKTWdrhaFErNvMV18Yl","function":{"name":"get_link_to_otel_semconv","arguments":"{\"semconv\":\"GenAI\"}"},"type":"function"}]}` |
+     | Event body (without content) | `{"index":0,"finish_reason":"tool_calls","message":{"tool_calls":[{"id":"call_VSPygqKTWdrhaFErNvMV18Yl","function":{"name":"get_link_to_otel_semconv"},"type":"function"}]}` |
+
+2. Application executes the tool call. Application may create span which is not covered by this semantic convention.
```

Comment:
> Should we make any relationship note about instrumentation of user-provided tool calls? For example, the span should be a child of the LLM one and it should have an attribute of the tool_call_id. I think saying that spans should be parented to the LLM if done, is reasonable to say (in the spec, not just hinted in an example)

maybe, but this will be out of scope of this PR - it currently only specifies what's in the prompt and completion events. Please create an issue if you'd be interested.; PRs are also appreciated.